### DISCUSSION TITLE
Help using `--reference-doc`

### DISCUSSION THREAD

**Discussion by souovan** [jgm/pandoc#9763] [Q&A] [answered]
Hi y'all i have the following use case:

i need to use a **reference.docx** file as template with `--reference-doc` while converting a `.md` file to output a `.docx` file, but i need that the `output.docx` file keep some pages unchanged from the **reference.docx** file, i appreciate any help !

**Comment by jgm** [comment, marked as answer]
Sorry, that's not possible.  The only thing pandoc carries over from the reference.docx are styles.

**Reply by souovan** [reply]
Thank you for your reply ! Is there any plan to implement such feature ?

**Reply by BrianMakes** [reply]
I was also wondering if there were plans to add to the functionality of docx templates.

**Reply by jgm** [reply]
I've recently added a feature that allows you to use an openxml template to customize the content of the page.  This will be in the next release.

**Reply by souovan** [reply]
Hi @jgm is this feature already implemented ? if yes could you tell me where i can find an example or the docs for it ?

**Reply by jgm** [reply]
We don't really have detailed documentation for it.  You can do `pandoc -D openxml` to get a copy of the default template, which you can modify and then pass to pandoc using `--template`.  AS for how to modify it, well, you need to examine OpenXML documentation or study examples produced by Word.  It's not an easy format.

**Reply by rnwst** [reply]
@souovan -- I recently wrote an [article](https://rnwest.engineer/auto-generate-docx-with-pandoc/) about creating DOCX files with pandoc. It includes some details on working with openxml templates and the reference-doc, which you may find helpful.

**Reply by jgm** [reply]
That's really useful. I will add a link to it on the [wiki](https://github.com/jgm/pandoc/wiki/Documentation-and-Tutorials).

**Reply by souovan** [reply]
> We don't really have detailed documentation for it. You can do `pandoc -D openxml` to get a copy of the default template, which you can modify and then pass to pandoc using `--template`. AS for how to modify it, well, you need to examine OpenXML documentation or study examples produced by Word. It's not an easy format.

Thanks!! but i tried and get

```bash
$ pandoc -D openxml
Could not find data file /usr/share/pandoc-3.1.3/data/templates/default.openxml
```

My pandoc version is

```bash
$ pandoc -v                                                                                                                                     97 ↵
pandoc 3.1.3
Features: -server +lua
Scripting engine: Lua 5.4
User data directory: /home/van/.local/share/pandoc
Copyright (C) 2006-2023 John MacFarlane. Web: https://pandoc.org
This is free software; see the source for copying conditions. There is no
warranty, not even for merchantability or fitness for a particular purpose.
```

**Reply by TomBener** [reply]
> I recently wrote an [article](https://rnwest.engineer/auto-generate-docx-with-pandoc/) about creating DOCX files with pandoc. It includes some details on working with openxml templates and the reference-doc, which you may find helpful.

Thanks for your great article and repo, West! I have tested it following your guidance, but I encountered PowerShell issues on my Mac. As a result, I created a [Bash-based repo](https://github.com/TomBener/pandoc-docx). Hope it helps for folks who use Mac.

**Reply by jidanni** [reply]
@rnwst in https://rnwest.engineer/auto-generate-docx-with-pandoc/
must not be using Debian, where all I get is
$ pandoc -o template.openxml --print-default-template=openxml
Could not find data file /usr/share/pandoc/data/templates/default.openxml

OK, I found it.
https://github.com/rnwst/pandoc-docx-tools/blob/master/template.openxml

**Comment by tarleb** [comment]
> I would need to review the full release notes, but I am fairly sure the openxml template was implemented after 3.1.3

Indeed, it was implemented in 3.2.1.
